core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,24 +22,48 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
       'siteAnalyticsService', 'UrlInterpolationService',
       function(
-          $scope, $window, oppiaHtmlEscaper, ExplorationEmbedButtonService,
+          $scope, $window, $injector, oppiaHtmlEscaper,
           siteAnalyticsService, UrlInterpolationService) {
-        $scope.explorationId = $scope.getExplorationId();
+        $scope.registerShareEvent = null;
 
-        $scope.registerShareExplorationEvent = function(network) {
-          siteAnalyticsService.registerShareExplorationEvent(network);
-        };
+        if ($scope.shareType === 'exploration') {
+          $scope.explorationId = $scope.getExplorationId();
+          var ExplorationEmbedButtonService = $injector.get(
+            'ExplorationEmbedButtonService');
 
-        $scope.showEmbedExplorationModal = (
-          ExplorationEmbedButtonService.showModal);
+          $scope.activityType = 'explore';
+          $scope.activityId = $scope.explorationId;
+
+          $scope.registerShareEvent = function(network) {
+            siteAnalyticsService.registerShareExplorationEvent(network);
+          };
+
+          $scope.showEmbedExplorationModal = (
+            ExplorationEmbedButtonService.showModal);
+        } else if ($scope.shareType === 'collection') {
+          $scope.collectionId = $scope.getCollectionId();
+
+          $scope.activityType = 'collection';
+          $scope.activityId = $scope.collectionId;
+
+          $scope.registerShareEvent = function(network) {
+            siteAnalyticsService.registerShareCollectionEvent(network);
+          };
+        } else {
+          throw (
```

Comment:
That's a good question. I think either is fine, see [here](http://stackoverflow.com/questions/13294658/throw-errormsg-vs-throw-new-errormsg).